### PR TITLE
cl12: Compiler - updated extension list for compiler_defines_for_exensions

### PR DESCRIPTION
**Pull request by Jeremy-Kemp**
Updated the list of known extensions to include all extensions referenced in the OpenCL Extension Specification 1.2 - Revision 25 and OpenCL Specification 1.2 - Revision 19 documents.

Also removed extensions from later OpenCL versions.

### CODE REVIEW

File: test_conformance/compiler/test_compiler_defines_for_extensions.cpp [hunk 1]
```diff
@@ -44,15 +45,17 @@ const char *known_extensions[] = {
     "cl_khr_egl_image",
     "cl_khr_egl_event",
     "cl_khr_create_command_queue",
-    "cl_khr_priority_hints",
-    "cl_khr_throttle_hints",
-    "cl_khr_il_program",
```
Comment (alycm) [inline test_conformance/compiler/test_compiler_defines_for_extensions.cpp]: `cl_khr_il_program` is a valid OpenCL 1.2 extension.

Comment (Jeremy-Kemp) [reply test_conformance/compiler/test_compiler_defines_for_extensions.cpp]: Good spot, thanks. I've just commited it back in.

**Comment by bashbaug** [comment]
Especially now, since we have a "unified" extensions spec, shouldn't we include "all" extensions in this list?  A few of them would be a little bit odd to support on an OpenCL 1.x device, but many others would make sense.

In particular, we enable priority hints and throttle hints for some of our OpenCL 1.x devices.

**Comment by Jeremy-Kemp** [comment]
> Especially now, since we have a "unified" extensions spec, shouldn't we include "all" extensions in this list? A few of them would be a little bit odd to support on an OpenCL 1.x device, but many others would make sense.
> 
> In particular, we enable priority hints and throttle hints for some of our OpenCL 1.x devices.

My rationale here was to have the 1.2 conformance branch test for extensions that are defined in the 1.2 OpenCL specs. I haven't yet looked at the unified spec to see what, if any, changes there are regarding when and/or how extensions are supported at particular OpenCL versions. (Is [this](https://github.com/KhronosGroup/OpenCL-Docs) the unified document?)

Further to that, I used the fact that [Clang](https://github.com/llvm-mirror/clang/blob/release_80/include/clang/Basic/OpenCLExtensions.def) defines the minimum supported OpenCL version for each extension (admittedly, Clang has a non-exhaustive list of extensions at the moment, but a good chunk of them are present in the latest release) to inform how to fix this test.

I don't have a strong feeling either way with this one, I'm happy to list every possible khr here in the test, or keep it as a 1:1 mapping to the specific version of OpenCL that the conformance branch is testing against.

**Comment by Jeremy-Kemp** [comment]
>So long as we don't remove the priority hint and throttle hint extensions I'm good.  smiley

In that case, I think we should go the whole hog and have every extension in the test, rather than a scatter-gun approach of adding extensions when required on an ad-hoc basis.

If people agree, I'll commit a change to add every khr from the unified spec.

**Comment by kpet** [comment]
+1 on adding all extensions from the unified spec.

**Comment by Jeremy-Kemp** [comment]
Ok, I've updated the test to include all extensions - thanks for the feedback! :)

Once it's approved, I'll create some more requests for the other branches so that we've got matching behavior.

**Comment by bashbaug** [comment]
I believe that priority hints and throttle hints are API-only extensions, but otherwise LGTM as well.  Thanks!

**Comment by Jeremy-Kemp** [comment]
Apologies for that, I'll move them back to the API-only section of the array. Did not notice in my testing as was using an implementation that does not expose those two extensions.

**Comment by alycm** [comment]
Oh, I built and tested this locally.  It worked for me.

I would merge this, but I have the open question that now that this address the unified extension spec I think that the equivalent change could be made to the cl20_trunk, cl21_trunk and cl22_trunk branches?

**Comment by kpet** [comment]
@Jeremy-Kemp Thanks for the changes. Agree with @alycm, could you prepare PRs for the other branches please?

**Comment by Jeremy-Kemp** [comment]
@kpet what is the standard way for replicating changes onto other branches for CTS? Should I cherry-pick, merge or just create new commits?

**Comment by kpet** [comment]
@Jeremy-Kemp I would start with a cherry-pick and amend where needed.

**Comment by alycm** [comment]
Comment on https://github.com/KhronosGroup/OpenCL-CTS/pull/175#discussion_r275414924 where I realised that `cl_khr_spirv_no_integer_wrap_decoration` is missing from this list.

**Comment by alycm** [comment]
I've retested this, it looks good!  I'll merge this and it's siblings.

**Pull request merged**